packages/cli/src/cmds/runCommand.ts

```diff
@@ -2,28 +2,78 @@ import { AbortError, ValidationError } from './errors';
 import { verbose } from '../utils';
 import UI from './userInteraction';
 import chalk from 'chalk';
-import Yargs from 'yargs';
+import Yargs, { command } from 'yargs';
+import { ExitCode } from './record/types/exitCode';
+import Telemetry from '../telemetry';
+import { exit } from 'process';
 
-export default async function runCommand(fn: () => Promise<any>) {
-  return fn().catch((err) => {
+async function invokeCommand() {}
+
+function flushTelemetry(exitCode: ExitCode, err?: Error) {
+  Telemetry.flush(() => {
+    if (err) {
+      Yargs.exit(exitCode, err);
+    } else {
+      exit(0);
```

Comment:
I believe this should be calling `Yargs.exit`, rather than `process.exit`. Yargs is kind of particular/weird about exiting the process.

This is the source for the `Yargs.exit`: https://github.com/yargs/yargs/blob/2d1136d303ea805685a973ded62f52efd49b78b9/lib/yargs-factory.ts#L724 , which expects that `err` can be undefined. IIRC, I've seen problems when a bad declaration for `Yargs.exit` gets picked up from somewhere else (maybe `@types`?).

Reply:
The problem for me is that `err` must be an Error, but I don't have one:

*@types/yargs*
```
        /** Manually indicate that the program should exit, and provide context about why we wanted to exit. Follows the behavior set by `.exitProcess().` */
        exit(code: number, err: Error): void;
```

Reply:
Yeah, that's the declaration I was thinking of. As you can see in the yargs source I linked to, it's wrong.

Can you pass a `new Error()`?